packages/flame/lib/src/post_process/post_process.dart

```diff
@@ -0,0 +1,138 @@
+import 'dart:async';
+import 'dart:ui' as ui show Image;
+import 'dart:ui' hide Image;
+
+import 'package:flame/components.dart';
+import 'package:flame/post_process.dart';
+import 'package:meta/meta.dart';
+
+abstract class PostProcess {
+  PostProcess({double? pixelRatio})
+      : pixelRatio = pixelRatio ??
+            PlatformDispatcher.instance.views.first.devicePixelRatio;
+
+  double pixelRatio;
+
+  FutureOr<void> onLoad() {}
+
+  void update(double dt) {}
+
+  void Function(Canvas)? _renderTree;
+  void Function(PostProcessRenderContext?)? _updateContext;
+  Vector2? _size;
+
+  @internal
+  void render(
+    Canvas canvas,
+    Vector2 size,
+    void Function(Canvas) renderTree,
+    void Function(PostProcessRenderContext?) updateContext,
+  ) {
+    _renderTree = renderTree;
+    _updateContext = updateContext;
+    _size = size;
+
+    canvas.save();
```

Comment:
It seems we're doing `canvas.save()` and `canvas.restore()` twice,
The first one is here that I mentioned, the second one is inside the calling function on lines 64 and 70